snuba/datasets/events_processor_base.py

```diff
@@ -108,6 +112,32 @@ def extract_promoted_contexts(
     ) -> None:
         raise NotImplementedError
 
+    def __track_latency(
+        self, output: MutableMapping[str, Any], data: MutableMapping[str, Any]
+    ) -> None:
+        # XXX: For now this function is purely for debugging purposes. While
+        # it would be nice to have these kind of statistics for our ingest
+        # system I don't think this hack is the correct solution.
+        now = datetime.utcnow().timestamp()
+        if "timestamp" in output:
+            metrics.timing(
+                "process_timestamp_latency",
+                (now - output["timestamp"].timestamp()) * 1000,
+            )
```

Comment:
I think we have something more reliable for this:
https://github.com/getsentry/arroyo/blob/b0a28268925347645974ca5174dd0369ff58fd2a/arroyo/processing/strategies/batching.py#L139-L146

Which is the time between when the event was inserted in the Kafka topic (events) and the time we were writing to clickhouse. It is what we use for our alerts and SLO.

A metric on event time is not supposed to be useful as the event tiestamp is self reported by the user. It can be in the past or even in the future, so unrelated from our ingestion pipeline.